### DISCUSSION TITLE
How do I submit covered call or cash secured PUT using the  API ?

### DISCUSSION THREAD

**Discussion by snrsamy** [erdewit/ib_insync#349] [Q&A]
I am looking to develop a process to place orders using the API for "Covered Calls" and "Cash Secured Puts" using the API.
Is there an option in the API to place "Sell to Open" and "Buy to Close" Orders ?

**Comment by m12t** [comment]
With IBKR you don't need to "sell to open" or "buy to close". You would simply enter a sell order for a put. So long as you have the cash to cover the position, it would be a cash secured put. Likewise, if you own 100 shares of the underlying, you can place a SELL order for a call and it will be automatically considered to be a covered call and your margin requirements will reflect this.
